src/citrine/jobs/job.py

```diff
@@ -50,13 +77,37 @@ class JobStatusResponse(Resource['JobStatusResponse']):
 
     job_type = properties.String("job_type")
     """:str: the type of job for this status report"""
-    status = properties.String("status")
+    _status = properties.String("status")
     """:str: The status of the job. One of "Running", "Success", or "Failure"."""
     tasks = properties.List(Object(TaskNode), "tasks")
     """:List[TaskNode]: all of the constituent task required to complete this job"""
     output = properties.Optional(properties.Mapping(String, String), 'output')
     """:Optional[dict[str, str]]: job output properties and results"""
 
+    @property
+    def status(self) -> Union[JobStatus, str]:
+        """The last reported status of this particular task."""
+        if resolved := JobStatus.from_str(self._status, exception=False):
+            return resolved
+        else:
+            return self._status
+
+    @status.setter
+    def status(self, value):
+        if resolved := JobStatus.from_str(value, exception=False):
+            if resolved not in [JobStatus.RUNNING, JobStatus.SUCCESS, JobStatus.FAILURE]:
```

Comment:
Shouldn't this support any valid `JobStatus` value? If so, you can use `list(JobStatus)` to ensure it's resilient against typos and future code changes.

Reply:
Oh, I see. That's strange, that the `Response` can't take the other statuses. Not even `Pending`?